api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -139,9 +139,11 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 							Obs replacement = os.saveObs(o, changeMessage);
 							//The logic in saveObs evicts the old obs instance, so we need to update
 							//the collection with the newly loaded and voided instance
-							toRemove.add(o);
-							toAdd.add(os.getObs(o.getId()));
-							toAdd.add(replacement);
+							if(!replacement.equals(o)) {
```

Comment:
So when the edited obs is removed from the group members of the parent, the obsGroup field is made null. Does that mean obs_group_id column is made null for the old obs? When is it saved to the db? I see no save operation of the edited obs after the obs group is made null.
https://github.com/openmrs/openmrs-core/blob/2.0.0/api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java#L155